core/src/main/java/org/jdbi/v3/core/internal/AnnotationFactory.java

```diff
@@ -49,12 +49,15 @@ private AnnotationFactory() {}
                         annotationType.getSimpleName()));
         });
 
-        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
         Class<?>[] interfaces = {annotationType};
         InvocationHandler invocationHandler = getInvocationHandler(annotationType, values);
 
         @SuppressWarnings("unchecked")
-        T annotation = (T) Proxy.newProxyInstance(classLoader, interfaces, invocationHandler);
+        T annotation = (T) Proxy.newProxyInstance(
+                Optional.ofNullable(annotationType.getClassLoader())
```

Comment:
nitpick: I preferred this as a variable (maybe `closestClassLoader`?) since now this block of code is pretty dense and multi-layered.